sidechain/primitives/src/types/block.rs

```diff
@@ -49,31 +52,28 @@ pub struct SignedBlock {
 #[derive(PartialEq, Eq, Clone, Encode, Decode, Debug)]
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 pub struct Block {
-	pub block_number: BlockNumber,
-	pub parent_hash: H256,
+	/// Sidechain Header
+	pub header: Header,
+
 	pub timestamp: u64,
 	/// Parentchain header this block is based on
 	pub layer_one_head: H256,
-	pub shard_id: ShardIdentifier,
 	///  must be registered on layer one as an enclave for the respective shard
 	pub block_author: ed25519::Public,
+
 	pub signed_top_hashes: Vec<H256>,
 	// encrypted state payload
 	pub state_payload: Vec<u8>,
 }
 
-impl BlockT for Block {
-	type ShardIdentifier = H256;
+impl BlockTrait for Block {
+	type HeaderType = Header;
 
 	type Public = ed25519::Public;
 
-	///get block number
-	fn block_number(&self) -> BlockNumber {
-		self.block_number
-	}
-	/// get parent hash of block
-	fn parent_hash(&self) -> H256 {
-		self.parent_hash
+	/// get header reference
```

Comment:
```suggestion
	/// Get header of the block.
```
It's not a reference, or is it?